### DISCUSSION TITLE
part-ii/2013-86/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#6] [Question comments]
Part II 2013 Logic and Set Theory: Paper 3, Section II, G

https://questions.tripos.org/part-ii/2013-86/

**Comment by ArthurConmy** [comment]
$X \cup \{ \neg \psi \}$ is inconsistent since if it was consistent, then the given set would be inconsistent since we have $\phi, \phi \implies \psi$ and modus ponens so $\psi$, contradicting $\neg \psi$. Hence by compactness there's a finite subset of the set that's inconsistent. This must include $\phi$. Therefore there are $q_1, ... , q_n, q'_1 , ... , q'_m$ so that $\phi \implies S$, where

$$ S = \neg ( q_1 \wedge ... \wedge q_n \wedge (\neg q_1') \wedge ... \wedge ( \neg q'_m ) ) $$

but then by definition we should have $S \in X$ and hence $q$ can't have been a valid valuation, contradiction.  

For the final part, again use compactness. Since $X \cup \{ \neg \psi \}$ is inconsistent there's some finite subset of it that's inconsistent. Either $X$ is inconsistent, in which case $\phi$ is and therefore $\chi = \neg$ works. Alternatively, the finite subset that's inconsistent contains $\neg \psi$ and therefore we have some $x_1, ... , x_p$ so that $\phi \implies x_1 \wedge ... \wedge x_p$ and $x_1 \wedge ... \wedge x_p \implies \psi$.
